DEV-GUIDELINES.md

```diff
@@ -0,0 +1,187 @@
+Development Guidelines
+------------
+
+This document provides a list of rules and best practices followed by project Helidon.
+Please follow these rules when contributing to the project or when refactoring existing code.
+
+**There may be reasons to violate these guidelines - in such cases, this must be documented in code, as otherwise 
```

Comment:
How about "Exception to the rules describes in this document should be documented as well as possible".

Reply:
Replaced